components/Members/useMembers.tsx

```diff
@@ -210,28 +217,76 @@ export default function useMembers() {
     ]
   )
 
+  // Loop through Members list to get our delegates and their tokens
+  // Return a object of key: walletId and value: object of arrays for council/community tokenOwnerRecords.
+  const getDelegateWalletMap = (members: Array<Member>): Delegates => {
+    const delegateMap = {} as Delegates
+    members.forEach((member: Member) => {
+      if (member?.delegateWalletCouncil) {
+        const walletId = member?.delegateWalletCouncil.toBase58()
+        if (delegateMap[walletId]) {
+          const oldCouncilRecords = delegateMap[walletId].councilMembers || []
+
+          delegateMap[walletId] = {
+            ...delegateMap[walletId],
+            councilMembers: [...oldCouncilRecords, member],
+            councilTokenCount:
+              (delegateMap[walletId]?.councilTokenCount || 0) +
+              member.councilVotes.toNumber(),
+          }
+        } else {
+          delegateMap[walletId] = {
+            councilMembers: [member],
+            councilTokenCount: member.councilVotes
+              ? member.councilVotes.toNumber()
+              : 0,
+          }
+        }
+      }
+
+      if (member?.delegateWalletCommunity) {
+        const walletId = member?.delegateWalletCommunity.toBase58()
+        if (delegateMap[walletId]) {
+          const oldCommunityRecords =
+            delegateMap[walletId].communityMembers || []
+
+          delegateMap[walletId] = {
+            ...delegateMap[walletId],
+            communityMembers: [...oldCommunityRecords, member],
+            communityTokenCount:
+              (delegateMap[walletId]?.communityTokenCount || 0) +
+              member.communityVotes.toNumber(),
+          }
+        } else {
+          delegateMap[walletId] = {
+            communityMembers: [member],
+            communityTokenCount: member.communityVotes
+              ? member.communityVotes.toNumber()
+              : 0,
+          }
+        }
+      }
+    })
+
+    return delegateMap
+  }
+
   //Move to store if will be used more across application
   useEffect(() => {
     const handleSetMembers = async () => {
       let members = [...membersWithTokensDeposited]
+
       const councilMembers = await fetchCouncilMembersWithTokensOutsideRealm()
       const communityMembers = await fetchCommunityMembersATAS()
       members = matchMembers(members, councilMembers, 'council', true)
       members = matchMembers(members, communityMembers, 'community')
+
+      const delegateMap = getDelegateWalletMap(members)
+      setDelegates(delegateMap)
       setMembers(members)
     }
-    if (
-      realm?.pubkey &&
-      previousRealmPubKey !== realm?.pubkey.toBase58() &&
-      !realm?.account.config.useCommunityVoterWeightAddin
-    ) {
+    if (realm?.pubkey && previousRealmPubKey !== realm?.pubkey.toBase58()) {
       handleSetMembers()
```

Comment:
it would be good to separate members and delegates set functions because members are not visible in lock tokens dao yet and its very heavy request so we would not like to call it in locked daos.